packages/data/src/namespace-store/index.js

```diff
@@ -188,7 +189,10 @@ function mapSelectors( selectors, store, registry ) {
  */
 function mapActions( actions, store ) {
 	const createBoundAction = ( action ) => ( ...args ) => {
-		store.dispatch( action( ...args ) );
+		const result = store.dispatch( action( ...args ) );
+		if ( isPromise( result ) ) {
+			return result;
```

Comment:
Per https://github.com/WordPress/gutenberg/pull/14711#issuecomment-479994280 , we should document this return value somewhere. Auto-doc probably won't help us. Probably along the mention of:

>`store.dispatch( action: Object )`: Given an action object, calls the registered reducer and updates the state value.

https://github.com/WordPress/gutenberg/blob/master/packages/data/README.md

Reply:
I added a note in the js doc block for the publicly exposed dispatch api (see 93172cf)  as opposed to what you referenced here which is the internal redux dispatch on the returned store object.  I _think_ this is likely what you meant.  I'm struggling a bit with the wording for additional docs so help is appreciated but imo I think we should keep it fairly brief for now.